rex/utilities/hpc.py

```diff
@@ -0,0 +1,738 @@
+            job_id = int(job[keys.index(cls.QCOL_ID)])
+            queue_dict[job_id] = {k: job[i] for i, k in enumerate(keys)}
+
+        return queue_dict
+
+    @abstractmethod
+    def query_queue(self, job_name=None, user=None, qformat=None,
+                    skip_rows=None):
+        """Run the HPC queue command and return the raw stdout string.
+
+        Parameters
+        ----------
+        job_name : str | None
+            Optional to check the squeue for a specific job name (not limited
+            to the 8 shown characters) or None to show user's whole queue.
+        user : str | None
+            HPC username. None will get your username using getpass.getuser()
+        qformat : str | None
+            Queue format string specification. Changing this form the
+            default (None) could have adverse effects!
+        skip_rows : int | list | None
+            Optional row index values to skip.
+
+        Returns
+        -------
+        stdout : str
+            HPC queue output string. Can be split on line breaks to get list.
+        """
+
+    @property
+    def queue(self):
+        """Get the HPC queue parsed into dict format keyed by integer job id
+
+        Returns
+        -------
+        queue : dict
+            HPC queue parsed into dictionary format keyed by integer job id
+            with nested dictionary of job properties (queue printout columns).
+        """
+        if self._queue is None:
+            qstr = self.query_queue(user=self._user)
+            self._queue = self.parse_queue_str(qstr)
+
+        return self._queue
+
+    @property
+    def queue_job_names(self):
+        """Get a list of the job names in the queue"""
+        return [attrs[self.QCOL_NAME] for attrs in self.queue.values()]
+
+    @property
+    def queue_job_ids(self):
+        """Get a list of the job integer ids in the queue"""
+        return list(self.queue.keys())
+
+    def check_status(self, job_id=None, job_name=None):
+        """Check the status of an HPC job using the HPC queue.
+
+        Parameters
+        ----------
+        job_id : int | None
+            Job integer ID number (preferred input)
+        job_name : str
+            Job name string.
+
+        Returns
+        -------
+        status : str | NoneType
+            Queue job status str or None if not found.
+            SLURM status strings: PD, R, CG (pending, running, complete).
+            PBS status strings: Q, R, C (queued, running, complete).
+        """
+
+        status = None
+
+        if job_id is not None:
+            if int(job_id) in self.queue:
+                status = self.queue[int(job_id)][self.QCOL_STATUS]
+
+        elif job_name is not None:
+            if job_name in self.queue_job_names:
+                for attrs in self.queue.values():
+                    if attrs[self.QCOL_NAME] == job_name:
+                        status = attrs[self.QCOL_STATUS]
+                        break
+
+        else:
```

Comment:
what if job_id and job_name are given?

Reply:
job id is first in the if statement so it gets priority

Reply:
Sounds good